dotnet/src/SemanticKernel/SkillDefinition/NativeFunction.cs

```diff
@@ -543,25 +543,33 @@ private static (Func<ITextCompletion?, CompleteRequestSettings?, SKContext, Canc
             return async static (result, context) =>
             {
                 await ((ValueTask)ThrowIfNullResult(result)).ConfigureAwait(false);
-                return context;
+                return new FunctionResult(context);
             };
         }
 
         // SKContext, either synchronous (SKContext) or asynchronous (Task<SKContext> / ValueTask<SKContext>).
 
         if (returnType == typeof(SKContext))
```

Comment:
Should we have a similar `if` operator for 'FunctionResult'? or the case is already handled?

Reply:
Good idea. In this case it will be possible for plugin developers to reuse `FunctionResult` as return type for their functions. Will add that, thanks. Most probably we will have to remove `if` for `SKContext`, as we shouldn't encourage developers to use SKContext as return type.